### PR TITLE
Deprecate `dataframe.tseries.resample.getnanos`

### PR DESCRIPTION
**Pull request by scharlottej13**
- [X] addresses #8688 
- [X] Tests added / passed
- [X] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/tseries/tests/test_resample.py
Comment (jrbourbeau) [inline dask/dataframe/tseries/tests/test_resample.py:220]: The warning message is slightly different after adding `after_version=` to `_deprecated`

```suggestion
    with pytest.warns(FutureWarning, match="getnanos was deprecated"):
```

File: dask/dataframe/tseries/resample.py
Comment (jrbourbeau) [inline dask/dataframe/tseries/resample.py:13]: One small suggestion about adding the version of `dask` after which `getnanos` was deprecated. This will make the deprecation message `getnanos was deprecated after version 2022.02.0...` instead of just `getnanos is deprecated...`, which will both provide users with a bit more info as well as make it easier for maintainers to see exactly when `getnanos` was deprecated (otherwise we'll need to go look at the changelog to see what release included this PR)

```suggestion
@_deprecated(after_version="2022.02.0")
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @scharlottej13! One minor comment, but otherwise this looks great

**Comment by scharlottej13** [comment]
> Thanks @scharlottej13! One minor comment, but otherwise this looks great

adding in the Dask version makes sense, thanks James!

**Review by jrbourbeau — APPROVED** [review]
Thanks for the updates @scharlottej13  -- will merge after CI finishes up

**Review by jcrist — APPROVED** [review]

**Pull request merged**
